Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -0,0 +1,197 @@
+import Foundation
+
+/// Represents a MongoDB connection string.
+/// - SeeAlso: https://docs.mongodb.com/manual/reference/connection-string/
+public struct MongoConnectionString: Codable, LosslessStringConvertible {
+    /// Represents a connection string scheme.
+    public struct Scheme: LosslessStringConvertible, Equatable {
+        /// Indicates that this connection string uses the scheme `mongodb`.
+        public static let mongodb = Scheme(.mongodb)
+
+        /// Indicates that this connection string uses the scheme `mongodb+srv`.
+        public static let srv = Scheme(.srv)
+
+        /// Internal representation of a scheme.
+        private enum _Scheme: String {
+            case mongodb
+            case srv = "mongodb+srv"
+        }
+
+        private let _scheme: _Scheme
+
+        private init(_ value: _Scheme) {
+            self._scheme = value
+        }
+
+        /// LosslessStringConvertible` protocol requirements
+        public init?(_ description: String) {
+            guard let _scheme = _Scheme(rawValue: description) else {
+                return nil
+            }
+            self.init(_scheme)
+        }
+
+        public var description: String { self._scheme.rawValue }
+    }
+
+    /// A struct representing a network address, consisting of a host and port.
+    public struct HostIdentifier: Equatable, CustomStringConvertible {
+        private static func verifyPort(from: String, scheme: Scheme) throws -> UInt16? {
+            guard scheme != .srv else {
+                throw MongoError.InvalidArgumentError(
+                    message: "mongodb+srv with port number"
```

Comment:
Can we update this error message to be a bit more specific, e.g. `"a port cannot be specified in a mongodb+srv connection string"`

Reply:
done